src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -95,44 +91,55 @@
   [ruby-load-path :- [schema/Str]]
   (cons ruby-code-dir ruby-load-path))
 
-(defn prep-scripting-container
-  [scripting-container ruby-load-path gem-home]
-  ; Note, this behavior should remain consistent with new-main
-  (doto scripting-container
+(schema/defn ^:always-validate get-compile-mode :- RubyInstanceConfig$CompileMode
+  [config-compile-mode :- jruby-schemas/SupportedJRubyCompileModes]
+  (case config-compile-mode
+    :jit RubyInstanceConfig$CompileMode/JIT
+    :force RubyInstanceConfig$CompileMode/FORCE
+    :off RubyInstanceConfig$CompileMode/OFF))
+
+(schema/defn ^:always-validate init-jruby-config :- jruby-schemas/ConfigurableJRuby
+  "Applies configuration to a JRuby... thing.  See comments in `ConfigurableJRuby`
+  schema for more details."
+  [jruby-config :- jruby-schemas/ConfigurableJRuby
```

Comment:
Another thought that occurred to me is that we could just implement some new adapter Java classes that inherit from a new common interface that we'd create with the setters that we know that we'll need to use but that derive from the appropriate JRuby concrete implementations.  That way, we could just refer to the interface in the schema validation here.  Where this PR already introduces a derived class implementation for `ScriptingContainer`, maybe it wouldn't be too onerous to do a new one for `RubyInstanceConfig` as well.  WDYT?